### PR TITLE
Add `vitest` globals

### PR DESCRIPTION
**Pull request by yongholeeme**
Hello @sindresorhus! I'm opening a PR to add `vitest` globals.

A similar PR was previously opened in https://github.com/sindresorhus/globals/pull/210, but it was closed due to the package's management policy at that time (as discussed in https://github.com/sindresorhus/globals/issues/82). However, since ESLint no longer bundles globals and new environments are now being added to the globals package, I believe it's appropriate to open this request.

I've added these by referencing the official Vitest globals definition file: https://github.com/vitest-dev/vitest/blob/main/packages/vitest/globals.d.ts.

### CODE REVIEW

File: globals.json [hunk 1]
```diff
@@ -2656,6 +2656,24 @@
 		"uniq": false,
 		"which": false
 	},
+	"vitest": {
```
Comment (fisker) [inline globals.json:2659]: The json file is created by script, please store data in `data/*.js`

Comment (yongholeeme) [reply globals.json:2659]: Got it! [I'll move the changes to data/*.js](https://github.com/sindresorhus/globals/pull/274/files#diff-8f59a9b4760647ce4e81ec0af25a6ac8cccf1588decb56f3404b605a0f070a92) and run the build script instead.

I've updated the PR and force-pushed the changes.

**Comment by fisker** [comment]
As I tested, `vitest` package can't be imported, it seems there is no way to get those from `vitest` package, but `vitest/globals` contains types, is it possible to assert they are equal here https://github.com/sindresorhus/globals/blob/main/index.test-d.ts?

**Comment by yongholeeme** [comment]
> As I tested, vitest package can't be imported, it seems there is no way to get those from vitest package, but vitest/globals contains types, is it possible to assert they are equal here https://github.com/sindresorhus/globals/blob/main/index.test-d.ts?

@fisker, as you mentioned, `vitest/globals` contains types in [globals.d.ts](https://github.com/vitest-dev/vitest/blob/main/packages/vitest/globals.d.ts). However, since these are declared as global types, I think it's difficult to assert their equality.

**Comment by yongholeeme** [comment]
@fisker, I've updated the way of generating vitest globals by using `@vitest/eslint-plugin` (see https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/src/index.ts#L269-L285). However, it's not identical to [`vitest/globals`](https://github.com/vitest-dev/vitest/blob/main/packages/vitest/globals.d.ts). As this is the official package, I think this approach would be better for maintaining globals, and I'll create an issue to request alignment between them.

**Review by fisker — APPROVED** [review]

**Pull request merged**